.github/workflows/release.yml

```diff
@@ -4,46 +4,197 @@ on:
+
+      - id: publish
+        name: Publish release
+        uses: actions/github-script@v4
+        with:
+          script: |
+            const fs = require('fs');
+            const version = fs.readFileSync('version', {encoding: 'ascii'}).trim();
+            const source_filename = fs.readFileSync('source-filename', {encoding: 'ascii'}).trim();
+            const release_body = fs.readFileSync('release-body', {encoding: 'utf-8'});
+
+            const source = fs.readFileSync(source_filename);
+
+            const release = await github.repos.createRelease({
+                owner: context.repo.owner, repo: context.repo.repo,
+                tag_name: version, name: version, body: release_body, draft: true
+                });
+
+            /* We get a lot of connection reset errors... */
+            for (let retry = 0; retry < 5; retry++) {
+                try {
+                    const upload = await github.repos.uploadReleaseAsset({
+                        owner: context.repo.owner, repo: context.repo.repo, release_id: release.data.id,
+                        name: source_filename, headers: {'Content-Type': 'application/x-xz'}, data: source});
+                    break;
+                } catch (error) {
+                    console.log(`Failed to upload (status ${error.status})`);
+                    if (error.status != 500) {
+                        throw error;
+                    }
+                }
+            }
+
+            const update = await github.repos.updateRelease({
+              owner: context.repo.owner, repo: context.repo.repo, release_id: release.data.id,
+              draft: false});
+
+            core.setOutput('checksum', require('crypto').createHash('sha256').update(source).digest('hex'));
+            core.setOutput('download', update.data.assets[0].browser_download_url);
+            core.setOutput('filename', source_filename);
+            core.setOutput('version', version);
+
+    outputs:
+      checksum: ${{ steps.publish.outputs.checksum }}
+      download: ${{ steps.publish.outputs.download }}
+      filename: ${{ steps.publish.outputs.filename }}
+      version: ${{ steps.publish.outputs.version }}
+
+
+  guide:
+    needs: source
+    environment: website
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/cockpit-project/unit-tests
+      options: --user root
+    steps:
+      - name: Checkout website repository
+        uses: actions/checkout@v3
+        with:
+          path: website
+          repository: cockpit-project/cockpit-project.github.io
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+
+      - name: Download source release
+        run: curl -L -o '${{ needs.source.outputs.filename }}' '${{ needs.source.outputs.download }}'
+
+      - name: Verify checksum
+        run: echo '${{ needs.source.outputs.checksum }} ${{ needs.source.outputs.filename }}' | sha256sum -c
+
+      - name: Build guide
```

Comment:
I'd prefer keeping tools/release-guide, and just calling that from the workflow with passing the `outputs` variables. It's so much easier to locally run and debug a shell script than a workflow..

Reply:
Let's talk about this.

Reply:
We're going to handle this as a follow-up, possibly using the new unit tests container script from #17304.